### DISCUSSION TITLE
Lesson 4: installing yarn

### DISCUSSION THREAD

**Discussion by bergerson92** [smartcontractkit/full-blockchain-solidity-course-py#260] [Q&A] [answered]
Hey,

has anyone had the same issue?

when installing yarn (nom install --global yarn) I got a error message with "No permission/access".
Which I fixed with sudo chown -R $USER /usr/local/lib/node_modules found [here](https://flaviocopes.com/npm-fix-missing-write-access-error/).

Anyway I still get this error message:

> yarn@1.22.17 preinstall /usr/local/lib/node_modules/yarn
> :; (node ./preinstall.js > /dev/null 2>&1 || true)

npm ERR! code EACCES
npm ERR! syscall symlink
npm ERR! path ../lib/node_modules/yarn/bin/yarn.js
npm ERR! dest /usr/local/bin/yarnpkg
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, symlink '../lib/node_modules/yarn/bin/yarn.js' -> '/usr/local/bin/yarnpkg'
npm ERR!  [OperationalError: EACCES: permission denied, symlink '../lib/node_modules/yarn/bin/yarn.js' -> '/usr/local/bin/yarnpkg'] {
npm ERR!   cause: [Error: EACCES: permission denied, symlink '../lib/node_modules/yarn/bin/yarn.js' -> '/usr/local/bin/yarnpkg'] {
npm ERR!     errno: -13,
npm ERR!     code: 'EACCES',
npm ERR!     syscall: 'symlink',
npm ERR!     path: '../lib/node_modules/yarn/bin/yarn.js',
npm ERR!     dest: '/usr/local/bin/yarnpkg'
npm ERR!   },
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'symlink',
npm ERR!   path: '../lib/node_modules/yarn/bin/yarn.js',
npm ERR!   dest: '/usr/local/bin/yarnpkg'
npm ERR! }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

thanks in advance

**Comment by bergerson92** [comment, marked as answer]
solved it by adding "sudo" in front :) 
`sudo npm install --global yarn`

**Reply by cromewar** [reply]
I'm glad you solved it, however it's a bad practice to install any npm packages using root privileges due to security reasons I highly recommend to use npm via EVM, you can use this as guide:

for WSL2
https://docs.microsoft.com/en-us/windows/dev-environment/javascript/nodejs-on-wsl
for Native Ubuntu:
https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-ubuntu-20-04-es

**Reply by bergerson92** [reply]
Hey,

didn't know that thanks for the clarification and the links.
